app/Models/CustomMap.php

```diff
@@ -80,9 +80,26 @@ public function getBackgroundConfig(): array
         return $config;
     }
 
-    public function hasAccess(): bool
+    public function hasReadAccess(User $user): bool
     {
-        return false; // TODO calculate based on device access
+        $results = $this->query()->where('custom_map_id', $this->custom_map_id)->withCount([
```

Comment:
I don't think executing an SQL query here is the best idea.  How does this query scale, and how often is it called?

If you do end up just using this SQL, you should just use the scope below as it seems to match.

Maybe scaling back the sql can help:
```
foreach ($this->nodes()->pluck('device_id') as $device_id) {
    if (! Permissions::canAccessDevice($device_id, $user)) {
        return false;
    }
}

return true;
```

Reply:
I've updated the code to us Permissions::canAccessDevice() as suggested to avoid the extra SQL query